networkx/algorithms/hierarchy_coordinates.py

```diff
@@ -0,0 +1,1139 @@
+            nx.to_numpy_array(G),
+            num_thresholds,
+            threshold_distribution,
+        )
+        o = sum(
+            orderability(OG, CG) for CG, OG in zip(condensed_graphs, original_graphs)
+        )
+        return o / len(condensed_graphs)
+
+    if condensed_nx_graph is None:
+        condensed_nx_graph = weight_nodes_by_condensation(nx.condensation(G))
+    weights = nx.get_node_attributes(condensed_nx_graph, "weight").values()
+    total_acyclic_node_weight = sum(weights)
+    number_non_cyclic_nodes = sum(1 for wt in weights if wt == 1)
+    return number_non_cyclic_nodes / total_acyclic_node_weight
+
+
+@not_implemented_for("undirected")
+def _feedforwardness_iteration(G):
+    """Performs a single iteration of the feedforward algorithm described in [1]_
+
+    Parameters
+    ----------
+    G: NetworkX Graph
+        a directed, acyclic graph [1]_
+
+    Return
+    --------
+    g: float
+        sum of feedforwardness for a single Directed Acyclic Graph
+    num_paths: int
+        sum total paths considered
+
+    Notes
+    ______
+    g(G) = sum of F over all paths from maximal to minimal nodes
+
+    .. [1] "On the origins of hierarchy in complex networks."
+     Corominas-Murtra, Bernat, Joaquín Goñi, Ricard V. Solé, and Carlos Rodríguez-Caso,
+     Proceedings of the National Academy of Sciences 110, no. 33 (2013)
+
+    See Also
+    --------
+    feedforwardness
+    """
+    if not nx.is_directed_acyclic_graph(G):
+        raise nx.NetworkXError(
+            "G must be a directed acyclic graph for feedforwardness evaluation"
+        )
+
+    max_layer = max_min_layers(G, max_layer=True)
+    min_layer = max_min_layers(G, max_layer=False)
+    weights = nx.get_node_attributes(G, "weight")
+    g = 0
+    num_paths = 0
+    for max_node in max_layer:
+        for min_node in min_layer:
+            for path in nx.all_simple_paths(G, source=max_node, target=min_node):
+                # where each path calculation is F(path)
+                g += len(path) / sum([weights[node] for node in path])
+                num_paths += 1
+    return g, num_paths
+
+
+@not_implemented_for("undirected")
+def feedforwardness(DAG):
+    """Returns feedforwardness hierarchy coordinates, ~extent information passes vertically
+    (vs horizontally) within a hierarchy.
+
+    Based on a weighted sum which considers how much each path from every maximal node to every minimal node
+    traverses through cyclic (flat hierarchy) components. This calculation is then repeated for every subgraph
+    produced through maximal layer removal.
+
+    The original algorithm was given in "On the origins of hierarchy in complex networks."[1]_ and with details
+    given in their appendix.[2]_
+
+    Parameters
+    ----------
+    DAG: NetworkX Graph
+        A Directed, Acyclic Graph.
+
+    Return
+    -------
+    float:
+        feedforwardness, hierarchy coordinate
+
+    Examples
```

Comment:
Should this example have some text describing what is going on?
- The matrix is hard to understand as a graph -- perhaps create a network graph by adding weighted edges?
- What are the 8 graphs that are produced and how are they related to the original?

Maybe there should be a big picture paragraph somewhere in this docstring to explain what this function does to a graph.

Reply:
I've simplified the example to just the infographic, without weighed edges; perhaps that better conveys the essence of it. 
The 8 graphs from the old example result from the default 8 (linear) thresholds produced from `node_weighted_condense`, a lá
```
nx_graphs = [nx.from_numpy_matrix(np.where(A > threshold, 1, 0), create_using=nx.DiGraph) for threshold in thresholds]
```
 I'd add some more big picture explanation, though I'm not sure where's appropriate.

Reply:
Add big picture description everywhere you can. It's really needed in this module. Users come to the docs via google searches so you can't tell where they will start. You need to make every doc_string be good enough to be the first doc_string people see.  If you end up repeating ideas in many functions, use the same language in each so the user recognizes that it is the same ideas as they saw before -- but if its the first time they understand what's going on.
:}